Sources/NIOPosix/Bootstrap.swift

```diff
@@ -482,6 +482,484 @@ public final class ServerBootstrap {
+        serverBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark?,
+        childBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark?,
+        isChildChannelOutboundHalfClosureEnabled: Bool,
+        _ register: @escaping Register
+    ) async throws -> NIOAsyncChannel<NIOAsyncChannel<ChildChannelInbound, ChildChannelOutbound>, Never>  {
+        let eventLoop = self.group.next()
+        let childEventLoopGroup = self.childGroup
+        let serverChannelOptions = self._serverChannelOptions
+        let serverChannelInit = self.serverChannelInit ?? { _ in eventLoop.makeSucceededFuture(()) }
+        let childChannelInit = self.childChannelInit
+        let childChannelOptions = self._childChannelOptions
+        
+        let serverChannel = try makeServerChannel(eventLoop as! SelectableEventLoop, childEventLoopGroup, self.enableMPTCP)
+        
+        return try await eventLoop.submit {
+            serverChannelOptions.applyAllChannelOptions(to: serverChannel).flatMap {
+                serverChannelInit(serverChannel)
+            }.flatMap {
+                do {
+                    try serverChannel.pipeline.syncOperations.addHandler(
+                        AcceptHandler(childChannelInitializer: childChannelInit, childChannelOptions: childChannelOptions),
+                        name: "AcceptHandler"
+                    )
+
+                    // We are wrapping the inbound channels into `NIOAsyncChannel`s with the transformation
+                    // closure of the `NIOAsyncChannel` that allows us to wrap them without adding
+                    // wrapping/unwrapping handlers to the pipeline.
+                    let asyncChannel = try NIOAsyncChannel<NIOAsyncChannel<ChildChannelInbound, ChildChannelOutbound>, Never>(
+                        synchronouslyWrapping: serverChannel,
+                        backpressureStrategy: serverBackpressureStrategy,
+                        transformationClosure: { channel in
+                            try NIOAsyncChannel(
+                                synchronouslyWrapping: channel,
+                                backpressureStrategy: childBackpressureStrategy,
+                                isOutboundHalfClosureEnabled: isChildChannelOutboundHalfClosureEnabled,
+                                inboundType: ChildChannelInbound.self,
+                                outboundType: ChildChannelOutbound.self
+                            )
+                        }
+
+                    )
+                    return register(eventLoop, serverChannel).map { asyncChannel }
+                } catch {
+                    return eventLoop.makeFailedFuture(error)
+                }
+            }.flatMapError { error in
+                serverChannel.close0(error: error, mode: .all, promise: nil)
+                return eventLoop.makeFailedFuture(error)
+            }
+        }.flatMap {
+            $0
```

Comment:
I think if you use `eventLoop.flatSubmit` you can drop this

Reply:
Where would I use the `flatSubmit` here?

Reply:
Instead of the `submit` on L699